presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -1180,7 +1180,9 @@ public static long widthBucket(@SqlType(StandardTypes.DOUBLE) double operand, @S
             index = (lower + upper) / 2;
             bin = DOUBLE.getDouble(bins, index);
 
-            checkCondition(isFinite(bin), INVALID_FUNCTION_ARGUMENT, format("Bin value must be finite, got %s", bin));
+            if (!isFinite(bin)) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Bin value must be finite, got %s", bin));
```

Comment:
nit: `format` is not necessary here, just `"Bin value must be finite, got " + bin`